src/containers/IntentService.jsx

```diff
@@ -36,23 +37,27 @@ export default class IntentService extends Component {
           disableSuccessTimeout: !!data.disableSuccessTimeout
         })
 
-        if (!data || !data.slug) {
+        if (!data) {
           throw new Error('Unexpected data from intent')
         }
 
-        return this.store.fetchKonnectorInfos(data.slug)
+        if (data.slug) {
+          return this.store.fetchKonnectorInfos(data.slug).then(konnector => [konnector])
+        } else if (data.dataType) {
+          return this.store.findByDataType(data.dataType)
+        }
       })
-      .then(konnector => {
-        if (!konnector) {
+      .then(konnectorsList => {
```

Comment:
Reminder for later: prefer `konnectorList` over `konnectorsList`, see https://forum.wordreference.com/threads/item-list-or-items-list.2236258/